prow/gangway/gangway.proto

```diff
@@ -0,0 +1,129 @@
+syntax = "proto3";
+
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "k8s.io/test-infra/prow/gangway";
+
+service Prow {
+  // FIXME: In the future we can just return a unique token (only), in the same
+  // way that GCB returns immediately with the globally-unique BuildId. That is,
+  // in the future the response will be a union of either the full JobExecution
+  // message or a single JobExecutionToken (string). See
+  // https://docs.google.com/document/d/1v77jp1Nb5C2C2-PdV02SGViO9CyZ9SvNxCPOHyIUQeo/edit#bookmark=id.q68srxklvpt4.
+  rpc CreateJobExecution(CreateJobExecutionRequest) returns (JobExecution) {
+    option (google.api.http) = {
+      post: "/v1/executions/{job_name}"
+      body: "*"  // See
+                 // https://cloud.google.com/endpoints/docs/grpc/transcoding#use_wildcard_in_body
+    };
+  }
+  rpc GetJobExecution(GetJobExecutionRequest) returns (JobExecution) {
+    // Client example:
+    //   curl http://DOMAIN_NAME/v1/executions/1
+    option (google.api.http) = {
+      get: "/v1/executions/{id}"
+    };
+  }
+  rpc ListJobExecutions(ListJobExecutionsRequest) returns (JobExecutions) {
+    // Client example:
+    //   curl
+    //   http://DOMAIN_NAME/v1/executions?job_name=my-prow-job&status=SUCCESS
+    option (google.api.http) = {
+      get: "/v1/executions"
+    };
+  }
+}
+
+message CreateJobExecutionRequest {
+  string job_name = 1;
+  JobExecutionType job_execution_type = 2;
+  Refs refs = 3;
+  PodSpecOptions pod_spec_options = 4;
+}
+
+message PodSpecOptions {
+  map<string, string> envs = 1;
+  map<string, string> labels = 2;
+  map<string, string> annotations = 3;
+}
+
+/* Look up a single Prow Job execution. */
+message GetJobExecutionRequest {
+  string id = 1;
+}
+
+/* Look up all Prow Job executions that match all fields given here. */
+message ListJobExecutionsRequest {
```

Comment:
For the Get and List request payloads do we plan to have separate `message` structs for different types of request parameters or will we just add optional fields?

e.g. We'll later want to be able to list all the job executions for a given PR (and maybe specific commit). Do you plan to add a separate request payload for this or augment `ListJobExecutionsRequest`?

I'm leaning toward the latter, but I ask this because if the former is expected we probably want more precise naming.

Reply:
I am also leaning toward the latter, mainly because it appears to be the simplest thing to do.